planning/behavior_path_planner/src/utils/start_goal_planner_common/utils.cpp

```
@@ -0,0 +1,111 @@
+    std::make_shared<EgoPredictedPathParams>(goal_planner_params->ego_predicted_path_params);
+}
+
+void updateSafetyCheckParams(
+  std::shared_ptr<SafetyCheckParams> & safety_check_params,
+  const std::shared_ptr<StartPlannerParameters> & start_planner_params)
+{
+  safety_check_params =
+    std::make_shared<SafetyCheckParams>(start_planner_params->safety_check_params);
+}
+
+void updateSafetyCheckParams(
+  std::shared_ptr<SafetyCheckParams> & safety_check_params,
+  const std::shared_ptr<GoalPlannerParameters> & goal_planner_params)
+{
+  safety_check_params =
+    std::make_shared<SafetyCheckParams>(goal_planner_params->safety_check_params);
+}
+
+void updateObjectsFilteringParams(
+  std::shared_ptr<ObjectsFilteringParams> & objects_filtering_params,
+  const std::shared_ptr<StartPlannerParameters> & start_planner_params)
+{
+  objects_filtering_params =
+    std::make_shared<ObjectsFilteringParams>(start_planner_params->objects_filtering_params);
+}
+
+void updateObjectsFilteringParams(
+  std::shared_ptr<ObjectsFilteringParams> & objects_filtering_params,
+  const std::shared_ptr<GoalPlannerParameters> & goal_planner_params)
+{
+  objects_filtering_params =
+    std::make_shared<ObjectsFilteringParams>(goal_planner_params->objects_filtering_params);
+}
+
+void updatePathProperty(
+  std::shared_ptr<EgoPredictedPathParams> & ego_predicted_path_params,
+  const std::pair<double, double> & pairs_terminal_velocity_and_accel)
+{
+  ego_predicted_path_params->target_velocity = pairs_terminal_velocity_and_accel.first;
+  ego_predicted_path_params->acceleration = pairs_terminal_velocity_and_accel.second;
+}
+
+std::pair<double, double> getPairsTerminalVelocityAndAccel(
+  const std::vector<std::pair<double, double>> & pairs_terminal_velocity_and_accel,
+  const size_t current_path_idx)
+{
+  if (pairs_terminal_velocity_and_accel.size() <= current_path_idx) {
+    std::cerr << "pair_terminal_velocity_and_accel is not given" << std::endl;
```

this may be noisy?

---

I don't expect current_idx is bigger than the size of pairs_terminal_velocity_and_accel.size() so actually want to throw an exception but I'm pretty sure this expectation is not ensured enough in the current temporary implementation.
But to avoid, noisy output I will comement out this line.

---

I removed this debug code in https://github.com/autowarefoundation/autoware.universe/pull/4804/commits/5a8ea3575527144c54b51989cb5f372b4dd7c34e